x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -276,6 +282,20 @@ public InferenceFieldMetadata getMetadata(Set<String> sourcePaths) {
         return new InferenceFieldMetadata(name(), fieldType().inferenceId, copyFields);
     }
 
+    @Override
+    public Object getOriginalValue(Map<String, Object> sourceAsMap) {
+        Object fieldValue = sourceAsMap.get(name());
+        if (fieldValue == null) {
+            return null;
+        } else if (fieldValue instanceof Map<?, ?> == false) {
+            // Don't try to further validate the non-map value, that will be handled when the source is fully parsed
+            return fieldValue;
+        }
+
+        Map<String, Object> fieldValueMap = XContentMapValues.nodeMapValue(fieldValue, "Field [" + name() + "]");
+        return XContentMapValues.extractValue(TEXT_FIELD, fieldValueMap);
+    }
```

Comment:
This method performs a partial parse of the provided source to get the original value